cpu/nrf52/ldscripts/nrf52832xxaa.ld

```diff
@@ -18,10 +18,23 @@
  * @}
  */
 
+rom_length = 512K;
+
+INCLUDE multislot.ld
+
 MEMORY
 {
-    rom (rx)    : ORIGIN = 0x00000000, LENGTH = 512K
-    ram (rwx)   : ORIGIN = 0x20000000, LENGTH = 64K
+    rom (rx)    : ORIGIN = 0x00000000 + boot_offset, LENGTH = rom_length
+    ram (rwx)   : ORIGIN = 0x20000000,               LENGTH = 64K
+}
+
+SECTIONS
+{
+    /* Image headers for MCUBoot */
+    .imghdr (NOLOAD):
+    {
+      . = . + img_hdr;
+    } > rom
```

Comment:
OK I understood your point, will change asap.

> This is my most concern. If the header is part of the image, the complexity of signing images increases a lot. How's the signature generated? Is the header part of it? We'd also basically force usage of an ELF aware image signing tool, instead of simply cat'ing header and elf together.

This is different on MCUBoot, on which the signatures and hashes are part of a TLV, at the end of the image, so no problem at this point. However, if we want to reuse this code for the RIOT bootloader then yes, this is a big concern (and I want to reuse this of course).


>the debugger knows the image start regardless of the image hdr defined in the linker address

This depends on how you start the debugger. In my very short experience, what I do is to first launch the debugger for the bootloader, as this is anyways the first piece of code to be executed. Then you need to know where the `.text` section for the next ELF file begins, in order to load it into the right address, which is not automatically read by the debugger. By having explicitly indicated the image headers, I was aware on which part the program begins, and so load the right address. Of course, this can be deduced from the text part, but you could wonder why the program starts at a "strange" address. Anyways, we can certainly live without such information. 

> the debugger doesn't mention the image header anywhere, regardless of it being defined in the linker script or not

Completely agree, but it was just for the case I explained above.